### PR TITLE
Feature banner helper

### PR DESCRIPTION
**Pull request by hannes-steffenhagen-diffblue**
This adds a small helper function, to help with writing banners like

```
**              this             **
```
which we do for our tools. Also updates all our help messages to use the helper.

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by smowton** [comment]
While you're at it, maybe fix the copyright dates?

**Review by NlightNFotis — APPROVED** [review]
Looks good to me.

**Comment by hannes-steffenhagen-diffblue** [comment]
@smowton I purposefully avoided messing with the copyright notices in any way because that's an issue I'd rather not deal with, if it was determined that they need changing I'd rather have that done separate from this work

File: src/util/parse_options.cpp [hunk 1]
```diff
@@ -116,19 +117,23 @@ int parse_options_baset::main()
...
+  auto const fill_left = fill - fill_right;
+  return border + std::string(fill_left, ' ') + text +
+         std::string(fill_right, ' ') + border;
+}
+
 std::string
 banner_string(const std::string &front_end, const std::string &version)
 {
+  static_assert(CHAR_BIT == 8,
+                "we do make the fairly safe assumption of 8 bit bytes");
```
Comment (tautschnig) [inline src/util/parse_options.cpp]: #917 :-)

**Comment by NlightNFotis** [comment]
Basically I agree with Hannes on this one: as a PR that just reformats the strings, it should only do just that. Changing the content of the strings, however trivial, in my opinion belongs to another PR.

File: unit/Makefile [hunk 2]
```diff
@@ -56,6 +56,7 @@ SRC += analyses/ai/ai.cpp \
        util/message.cpp \
        util/optional.cpp \
        util/optional_utils.cpp \
+			 util/parse_options.cpp \
```
Comment (tautschnig) [inline unit/Makefile]: Space vs tabs?

File: unit/util/parse_options.cpp [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#include <testing-utils/use_catch.h>
+#include <util/parse_options.h>
+
+TEST_CASE("align_center_with_border", "[core][util]") {
```
Comment (tautschnig) [inline unit/util/parse_options.cpp]: `{` on new line?

**Review by tautschnig — APPROVED** [review]

**Review by danpoe — APPROVED** [review]
LGTM

File: src/util/parse_options.cpp (hunk 1 above)
Comment (hannes-steffenhagen-diffblue) [reply src/util/parse_options.cpp]: Fair enough, I'll remove it. In this case the worst that can happen is odd numbers in the `-bits` section anyway :woman_shrugging:

File: unit/Makefile (hunk 2 above)
Comment (hannes-steffenhagen-diffblue) [reply unit/Makefile]: :woman_facepalming:

File: src/util/parse_options.cpp (hunk 1 above)
Comment (tautschnig) [reply src/util/parse_options.cpp]: Oh, sorry, I did not necessarily want you to feel compelled to remove it. Was just meaning to say that there is existing work to clean this up.

**Review by martin-cs — APPROVED** [review]
Couldn't we use left-pad :-D

**Comment by hannes-steffenhagen-diffblue** [comment]
Slight change: There was a potential underflow bug if the total length of the banner text exceeded the expected total length of the banner. The "fill" now can't be lower than 0.

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 55a8e6155e61c29fd4bf49c9a7f3d18f8e9b1e43).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/98597198
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Comment by hannes-steffenhagen-diffblue** [comment]
test-gen failure appears to be unrelated to this PR

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: b7a33ba5ce24173899f3efc3f248d8ab5fde394e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/98634346
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**